modules/indexer/issues/meilisearch/meilisearch.go

```diff
@@ -70,12 +70,19 @@ func (b *Indexer) Delete(_ context.Context, ids ...int64) error {
 
 // Search searches for issues by given conditions.
 // Returns the matching issue IDs
-func (b *Indexer) Search(ctx context.Context, keyword string, repoIDs []int64, limit, start int) (*internal.SearchResult, error) {
+func (b *Indexer) Search(ctx context.Context, keyword string, repoIDs []int64, limit, start int, state string) (*internal.SearchResult, error) {
 	repoFilters := make([]string, 0, len(repoIDs))
 	for _, repoID := range repoIDs {
 		repoFilters = append(repoFilters, "repo_id = "+strconv.FormatInt(repoID, 10))
 	}
 	filter := strings.Join(repoFilters, " OR ")
+	if state == "open" || state == "closed" {
+		if filter != "" {
+			filter = "(" + filter + ") AND state = " + state
+		} else {
+			filter = "state = " + state
+		}
+	}
```

Comment:
At the moment it is safe but there might be some changes cause SQL injection in the future for such code.

Reply:
And, how could the code work if `var state="open"`?

Then the SQL becomes `WHERE state = open`, but it should be `WHERE state = 'open'` (quoted) ?

Reply:
This isn't sql, but instead a custom query language by meilisearch https://www.meilisearch.com/docs/reference/api/search#filter no quotes needed (but can be optionally used)

Reply:
Ohhh, just realized that the file is `meilisearch.go`

Reply:
no worries ;)